idaes/core/ui/fsvis/static/js/stream_table.js

```diff
@@ -15,6 +25,61 @@ export class StreamTable {
         // Clear the table
         $("#hide-fields-list").empty();
         $("#stream-table-data").empty();
+        // Clear list of existing variable types
+        this.existing_var_types = new Set();
+    }
+
+    fillVarTypesPanel() {
+        /** Create a panel for each variable type that exists in the Stream Table */
+        const var_types_panel = document.querySelector('#existing-variable-types');
+        const stream_table_class = 'streamtable-vartype-element';
+
+        // Adding header
+        if (this.existing_var_types.size > 0) {
+            const header_vartype = document.createElement('p');
+            header_vartype.innerHTML = 'Annotated Variable Types:';
+            header_vartype.className = stream_table_class;
+            var_types_panel.appendChild(header_vartype);
+        }
+
+        // Adding each type
+        this.existing_var_types.forEach(var_type => {
+            const elem_vartype = document.createElement('span'); // Parent node
+            elem_vartype.className = stream_table_class;
+
+            // Create dot with the right color and the right variable type text
+            const elem_dot = document.createElement('span');
+            const elem_text = document.createElement('span');
+            elem_text.className = 'streamtable-vartype-text';
+            const var_type_class = `streamtable-vartype-${var_type}`;
+            switch (var_type) {
+                case this.UNFIXED:
+                    // This will execute once since this.existing_var_types is a set
+                    console.debug(`Unfixed variables don't have a visual indicator`);
+                    break;
+                case this.FIXED:
```

Comment:
Since all these cases do the same exact thing, you can have only one body:
```
case this.FIXED:
case this.PARAMETER:
case this.EXPRESSION:
    elem_dot.className = var_type_class;
    elem_dot.title = var_type;
    elem_text.innerHTML = var_type;
    break;
default: ..etc..
```